### PR TITLE
Interpolation functions implementation for missing data handling

### PR DESCRIPTION
**Pull request by agelloz**
New functions implementation:
- FillPrevious: fills missing data using LOCF (Last Obs. Carried Forward) method
- FillNext: fills missing data using NOCB (Next Obs. Carried Backward) method
- FillMean: fills missing data by calculating the average of the previous and the next observations
- FillLinear: fills missing data by calculating a linear value compared to a sorted reference column

New function NewRandomBow to generate a new random bow with or without missing data

Small refactoring of the package to improve user experience (to be continued...)

### CODE REVIEW

**Comment by agelloz** [comment]
New changes:
- FillLinear: added test isColSorted to enforce refCol to be sorted before using Filllinear
- FillLinear protection against convert error & isColSorted refacto

File: bowfill_test.go [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+
+func TestBow_FillIsColSorted(t *testing.T) {
+	var err error
+	intBobow, _ := NewBowFromRowBasedInterfaces([]string{"a", "b", "c", "d", "e"}, []Type{Int64, Int64, Int64, Int64, Int64}, [][]interface{}{
+		{-2, 1, nil, nil, -8},
+		{0, nil, 3, 4, 0},
+		{1, nil, nil, 120, nil},
+		{10, 4, 10, 10, -5},
+		{13, nil, nil, nil, nil},
+		{20, 6, 30, 400, -10},
```
[the remark this answers is not in the document]
Comment (nyhu) [reply bowfill_test.go]: use one bow, expected array and a dynamic subtest for loop with explicit col names

File: bowfill_test.go [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+		err = isColSorted(intBobow.(*bow), 3)
+		assert.Error(t, err)
+	})
+
+	t.Run("column e unsorted", func(t *testing.T) {
+		err = isColSorted(intBobow.(*bow), 4)
+		assert.Error(t, err)
+	})
+
+	t.Run("int64 random bow with missing data asc sorted", func(t *testing.T) {
```
Comment (nyhu) [inline bowfill_test.go]: this is testing the random

File: bowfill_test.go [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+
+// TODO: need to add more tests for FillLinear
+func TestBow_FillLinear(t *testing.T) {
+	holedInt, _ := NewBowFromRowBasedInterfaces([]string{"a", "b", "c", "d", "e"}, []Type{Int64, Int64, Int64, Int64, Int64}, [][]interface{}{
+		{-2, 1, nil, nil, -8},
+		{0, nil, 3, 4, 0},
+		{1, nil, nil, 120, nil},
+		{10, 4, 10, 10, -5},
+		{13, nil, nil, nil, nil},
+		{20, 6, 30, 400, -10},
```
Comment (nyhu) [inline bowfill_test.go]: add all empty use case

File: bowfill_test.go [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+	})
```
Comment (nyhu) [inline bowfill_test.go]: add first elem nil

Comment (nyhu) [inline bowfill_test.go]: add last elem nil

File: bowfill_test.go [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+		assert.Error(t, err)
+	})
+
+	t.Run("error unsorted refCol e", func(t *testing.T) {
+		filled, err := unsortedRefCol.FillLinear("e", "b")
+		assert.Nil(t, filled)
+		assert.Error(t, err)
+	})
+}
+
```
Comment (nyhu) [inline bowfill_test.go]: us a single bow for each type for each fill use case

**Pull request merged**